core/node/libp2p/routing.go

```diff
@@ -0,0 +1,108 @@
+package libp2p
+
+import (
+	"context"
+	"sort"
+
+	host "github.com/libp2p/go-libp2p-host"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	"github.com/libp2p/go-libp2p-pubsub"
+	namesys "github.com/libp2p/go-libp2p-pubsub-router"
+	record "github.com/libp2p/go-libp2p-record"
+	routing "github.com/libp2p/go-libp2p-routing"
+	routinghelpers "github.com/libp2p/go-libp2p-routing-helpers"
+	"go.uber.org/fx"
+
+	"github.com/ipfs/go-ipfs/core/node/helpers"
+	"github.com/ipfs/go-ipfs/repo"
+)
+
+type BaseIpfsRouting routing.IpfsRouting
+
+type Router struct {
+	routing.IpfsRouting
+
+	Priority int // less = more important
+}
+
+type p2pRouterOut struct {
+	fx.Out
+
+	Router Router `group:"routers"`
+}
+
+func BaseRouting(lc fx.Lifecycle, in BaseIpfsRouting) (out p2pRouterOut, dr *dht.IpfsDHT) {
```

Comment:
exported func BaseRouting returns unexported type libp2p.p2pRouterOut, which can be annoying to use